### ISSUE TITLE
Disallow invalid uses of `default` in rule schemas

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#78]
Ajv ignores `default` in rule schemas if they appear within `anyOf` or `oneOf` modifiers. It would be nice to have a rule that warns about these invalid schemas.

```js
module.exports = {
    meta: {
        schema: [
            {
                oneOf: [
                    { enum: ["foo", "bar"] },
                    {
                        type: "object",
                        properties: {
                            foo: {
                                type: "boolean",
                                default: true // invalid
                            }
                        }
                    }
                ]
            }
        ]
    }
    create(context) { /* ... */ }
};
```

Refs: https://github.com/eslint/eslint/pull/11427

**Comment by not-an-aardvark** [comment]
Closing in favor of https://github.com/eslint/eslint/issues/11473
